### PR TITLE
Use mouse to jump to any node(position) in  variation window

### PR DESCRIPTION
**Pull request by phiming**
#373 use mouse click to navigate in branch window just like in Sabaki

### CODE REVIEW

**Comment by zsalch** [comment]
@phiming 
Thank you for implementing this great feature.

Please help to check the following before the code review:

1. Code indentation uses 4 spaces
2. Variable names are as meaningful as possible
3. Remove unnecessary blank lines and useless code

more detail can refer to the review comment of @OlivierBlanvillain:
#364 #368

**Comment by phiming** [comment]
@zsalch 
OK,  for 
    1, 3,   I  will follow it.
    2,  I will add more comments and modifications to make code more readable.

File: src/main/java/featurecat/lizzie/gui/VariationTree.java [hunk 1]
```diff
@@ -13,12 +13,15 @@
     private int XSPACING;
     private int DOT_DIAM = 11; // Should be odd number
 
-    private ArrayList<Integer> laneUsageList;
-    private BoardHistoryNode curMove;
+    private ArrayList<Integer> laneUsageList, laneUsageList1;
```
Comment (zsalch) [inline src/main/java/featurecat/lizzie/gui/VariationTree.java]: The laneUsageList1 should be more sense.

File: src/main/java/featurecat/lizzie/gui/VariationTree.java [hunk 2]
```diff
@@ -13,12 +13,15 @@
...
+    private BoardHistoryNode curMove, curMove1;
+
+	private int posx1, posy1, width1, height1;   //Reserve a copy of varaition window axis
```
Comment (zsalch) [inline src/main/java/featurecat/lizzie/gui/VariationTree.java]: Please check the indent & variables sense.

File: src/main/java/featurecat/lizzie/gui/VariationTree.java [hunk 3]
```diff
@@ -114,6 +117,8 @@ public void draw(Graphics2D g, int posx, int posy, int width, int height) {
         YSPACING = (Lizzie.config.showLargeSubBoard() ? 20 : 30);
         XSPACING = YSPACING;
 
+	    posx1=posx;posy1=posy;width1=width;height1=height;  //backup for mouse click event
```
Comment (zsalch) [inline src/main/java/featurecat/lizzie/gui/VariationTree.java]: Please append spaces on both sides of the '=' and split the assignment statement to new lines.

File: src/main/java/featurecat/lizzie/gui/VariationTree.java [hunk 4]
```diff
@@ -142,4 +147,102 @@ public void draw(Graphics2D g, int posx, int posy, int width, int height) {
         }
         drawTree(g, posx + xoffset, curposy, 0, posy + height, node, 0,true);
     }
+	
+
```
Comment (zsalch) [inline src/main/java/featurecat/lizzie/gui/VariationTree.java]: Remove duplicate blank line.

File: src/main/java/featurecat/lizzie/gui/VariationTree.java [hunk 5]
```diff
@@ -142,4 +147,102 @@ public void draw(Graphics2D g, int posx, int posy, int width, int height) {
         }
         drawTree(g, posx + xoffset, curposy, 0, posy + height, node, 0,true);
     }
+	
+	 
+    //Clone from drawTree() method but draw nothing, just caculate (x,y)->BoardNode
+	public BoardHistoryNode inSubtree(int posx, int posy, int mouseX, int mouseY, int startLane, int maxposy, BoardHistoryNode startNode, int variationNumber)
+	{
```
Comment (zsalch) [inline src/main/java/featurecat/lizzie/gui/VariationTree.java]: The '{' should be in the end of line.

**Comment by phiming** [comment]
Thanks for your careful review.

I use vscode to write and debug those code. It looks good in vscode, but there are still some blanks and spaces I cant find. Is there any tools to check this automatically?

**Comment by featurecat** [comment]
I recommend using intellij or eclipse, they have auto formatters

intellij: ctrl alt L

eclipse: ctrl shift f

**Comment by featurecat** [comment]
VS code may have its own formatter. I don't know

**Comment by OlivierBlanvillain** [comment]
I'm actually thinking about setting up this mvn plugin to autoformat the code: https://github.com/google/google-java-format. It needs a tiny bit of work to prevent very large lines from becoming unreadable, but once I'm done with that we shouldn't have to review anything related to code format...

**Comment by OlivierBlanvillain** [comment]
@phiming I pushed a merge commit to your branch and I think it confused github, it doesn't show your commits here and won't let me reopen int :confused:... I opened #391 with your changed plus my merge commit, lets continue from there, sorry about that.

**Comment by OlivierBlanvillain** [comment]
I will try to review it ASAP :smile:!

**Pull request closed without merging**